From f0c6c8c99dc1ddc16d5f69304bc380cec24c5a26 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 24 Apr 2023 18:08:38 +0200
Subject: [PATCH 0614/1695] dts: stm32: add STM32MP21 pinctrl DT file

File is empty for the moment and will be filled with peripheral
pins nodes.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d070757d3b2f08e98155748dcf89eab8cda672b0 dts: stm32: add STM32MP21 pinctrl DT file]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I08e80d001c9d9e650f79302b6101fab7ef496d97
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361563
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp21-pinctrl.dtsi     | 10 +++
 core/arch/arm/dts/stm32mp21xxal-pinctrl.dtsi | 69 ++++++++++++++++++++
 core/arch/arm/dts/stm32mp21xxam-pinctrl.dtsi | 69 ++++++++++++++++++++
 core/arch/arm/dts/stm32mp21xxan-pinctrl.dtsi | 69 ++++++++++++++++++++
 core/arch/arm/dts/stm32mp21xxao-pinctrl.dtsi | 61 +++++++++++++++++
 5 files changed, 278 insertions(+)
 create mode 100644 core/arch/arm/dts/stm32mp21-pinctrl.dtsi
 create mode 100644 core/arch/arm/dts/stm32mp21xxal-pinctrl.dtsi
 create mode 100644 core/arch/arm/dts/stm32mp21xxam-pinctrl.dtsi
 create mode 100644 core/arch/arm/dts/stm32mp21xxan-pinctrl.dtsi
 create mode 100644 core/arch/arm/dts/stm32mp21xxao-pinctrl.dtsi

diff --git a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
new file mode 100644
index 000000000..ccfddd0d8
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) 2024, STMicroelectronics - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+#include <dt-bindings/pinctrl/stm32-pinfunc.h>
+
+&pinctrl {
+};
diff --git a/core/arch/arm/dts/stm32mp21xxal-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21xxal-pinctrl.dtsi
new file mode 100644
index 000000000..f3be620ce
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp21xxal-pinctrl.dtsi
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+&pinctrl {
+	gpioa: gpio@44240000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 0 16>;
+	};
+
+	gpiob: gpio@44250000 {
+		status = "okay";
+		ngpios = <13>;
+		gpio-ranges = <&pinctrl 1 17 3>, <&pinctrl 5 21 3>, <&pinctrl 9 25 7>;
+	};
+
+	gpioc: gpio@44260000 {
+		status = "okay";
+		ngpios = <14>;
+		gpio-ranges = <&pinctrl 0 32 14>;
+	};
+
+	gpiod: gpio@44270000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 48 16>;
+	};
+
+	gpioe: gpio@44280000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 64 16>;
+	};
+
+	gpiof: gpio@44290000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 80 14>, <&pinctrl 15 95 1>;
+	};
+
+	gpiog: gpio@442a0000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 96 6>, <&pinctrl 7 103 9>;
+	};
+
+	gpioh: gpio@442b0000 {
+		status = "okay";
+		ngpios = <9>;
+		gpio-ranges = <&pinctrl 4 116 2>, <&pinctrl 7 119 7>;
+	};
+
+	gpioi: gpio@442c0000 {
+		status = "okay";
+		ngpios = <6>;
+		gpio-ranges = <&pinctrl 0 128 2>, <&pinctrl 4 132 3>, <&pinctrl 8 136 1>;
+	};
+};
+
+&pinctrl_z {
+	gpioz: gpio@46200000 {
+		status = "okay";
+		ngpios = <3>;
+		gpio-ranges = <&pinctrl_z 0 400 2>, <&pinctrl_z 3 403 1>;
+	};
+};
diff --git a/core/arch/arm/dts/stm32mp21xxam-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21xxam-pinctrl.dtsi
new file mode 100644
index 000000000..f3be620ce
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp21xxam-pinctrl.dtsi
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+&pinctrl {
+	gpioa: gpio@44240000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 0 16>;
+	};
+
+	gpiob: gpio@44250000 {
+		status = "okay";
+		ngpios = <13>;
+		gpio-ranges = <&pinctrl 1 17 3>, <&pinctrl 5 21 3>, <&pinctrl 9 25 7>;
+	};
+
+	gpioc: gpio@44260000 {
+		status = "okay";
+		ngpios = <14>;
+		gpio-ranges = <&pinctrl 0 32 14>;
+	};
+
+	gpiod: gpio@44270000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 48 16>;
+	};
+
+	gpioe: gpio@44280000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 64 16>;
+	};
+
+	gpiof: gpio@44290000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 80 14>, <&pinctrl 15 95 1>;
+	};
+
+	gpiog: gpio@442a0000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 96 6>, <&pinctrl 7 103 9>;
+	};
+
+	gpioh: gpio@442b0000 {
+		status = "okay";
+		ngpios = <9>;
+		gpio-ranges = <&pinctrl 4 116 2>, <&pinctrl 7 119 7>;
+	};
+
+	gpioi: gpio@442c0000 {
+		status = "okay";
+		ngpios = <6>;
+		gpio-ranges = <&pinctrl 0 128 2>, <&pinctrl 4 132 3>, <&pinctrl 8 136 1>;
+	};
+};
+
+&pinctrl_z {
+	gpioz: gpio@46200000 {
+		status = "okay";
+		ngpios = <3>;
+		gpio-ranges = <&pinctrl_z 0 400 2>, <&pinctrl_z 3 403 1>;
+	};
+};
diff --git a/core/arch/arm/dts/stm32mp21xxan-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21xxan-pinctrl.dtsi
new file mode 100644
index 000000000..f3be620ce
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp21xxan-pinctrl.dtsi
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+&pinctrl {
+	gpioa: gpio@44240000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 0 16>;
+	};
+
+	gpiob: gpio@44250000 {
+		status = "okay";
+		ngpios = <13>;
+		gpio-ranges = <&pinctrl 1 17 3>, <&pinctrl 5 21 3>, <&pinctrl 9 25 7>;
+	};
+
+	gpioc: gpio@44260000 {
+		status = "okay";
+		ngpios = <14>;
+		gpio-ranges = <&pinctrl 0 32 14>;
+	};
+
+	gpiod: gpio@44270000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 48 16>;
+	};
+
+	gpioe: gpio@44280000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 64 16>;
+	};
+
+	gpiof: gpio@44290000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 80 14>, <&pinctrl 15 95 1>;
+	};
+
+	gpiog: gpio@442a0000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 96 6>, <&pinctrl 7 103 9>;
+	};
+
+	gpioh: gpio@442b0000 {
+		status = "okay";
+		ngpios = <9>;
+		gpio-ranges = <&pinctrl 4 116 2>, <&pinctrl 7 119 7>;
+	};
+
+	gpioi: gpio@442c0000 {
+		status = "okay";
+		ngpios = <6>;
+		gpio-ranges = <&pinctrl 0 128 2>, <&pinctrl 4 132 3>, <&pinctrl 8 136 1>;
+	};
+};
+
+&pinctrl_z {
+	gpioz: gpio@46200000 {
+		status = "okay";
+		ngpios = <3>;
+		gpio-ranges = <&pinctrl_z 0 400 2>, <&pinctrl_z 3 403 1>;
+	};
+};
diff --git a/core/arch/arm/dts/stm32mp21xxao-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21xxao-pinctrl.dtsi
new file mode 100644
index 000000000..fd9f798af
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp21xxao-pinctrl.dtsi
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2024 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+&pinctrl {
+	gpioa: gpio@44240000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 0 12>, <&pinctrl 13 13 3>;
+	};
+
+	gpiob: gpio@44250000 {
+		status = "okay";
+		ngpios = <5>;
+		gpio-ranges = <&pinctrl 11 27 5>;
+	};
+
+	gpioc: gpio@44260000 {
+		status = "okay";
+		ngpios = <14>;
+		gpio-ranges = <&pinctrl 0 32 14>;
+	};
+
+	gpiod: gpio@44270000 {
+		status = "okay";
+		ngpios = <16>;
+		gpio-ranges = <&pinctrl 0 48 16>;
+	};
+
+	gpioe: gpio@44280000 {
+		status = "okay";
+		ngpios = <14>;
+		gpio-ranges = <&pinctrl 0 64 10>, <&pinctrl 11 75 3>, <&pinctrl 15 79 1>;
+	};
+
+	gpiof: gpio@44290000 {
+		status = "okay";
+		ngpios = <15>;
+		gpio-ranges = <&pinctrl 0 80 14>, <&pinctrl 15 95 1>;
+	};
+
+	gpiog: gpio@442a0000 {
+		status = "okay";
+		ngpios = <9>;
+		gpio-ranges = <&pinctrl 0 96 6>, <&pinctrl 7 103 1>, <&pinctrl 11 107 2>;
+	};
+
+	gpioh: gpio@442b0000 {
+		status = "okay";
+		ngpios = <7>;
+		gpio-ranges = <&pinctrl 4 116 2>, <&pinctrl 9 121 5>;
+	};
+
+	gpioi: gpio@442c0000 {
+		status = "okay";
+		ngpios = <3>;
+		gpio-ranges = <&pinctrl 0 128 2>, <&pinctrl 4 132 1>;
+	};
+};
-- 
2.47.2

